Plugins/Rename/Rename.cpp

```diff
@@ -0,0 +1,219 @@
+			pCreature->m_pStats->m_lsLastName = lastName->second;
+        }
+    }
+
+	return retVal;
+	
+}
+
+void Rename::UpdateName(CNWSCreature* targetObject)
+{
+    API::CServerExoApp* server = Globals::AppManager()->m_pServerExoApp;
+    std::vector<Types::PlayerID> playersToNotify;
+    
+    API::CExoLinkedListInternal* playerList = server->m_pcExoAppInternal->m_pNWSPlayerList->m_pcExoLinkedListInternal;
+    
+    for (API::CExoLinkedListNode* head = playerList->pHead; head; head = head->pNext)
+    {
+        API::CNWSClient* client = reinterpret_cast<API::CNWSClient*>(head->pObject);
+        if (client)
+        {
+            playersToNotify.emplace_back(client->m_nPlayerID); //gather a list of players to iterate through and notify of the name change
+        }
+    }
+
+    API::CNWSMessage* message = server->m_pcExoAppInternal->m_pMessage;
+
+    for (auto& pid : playersToNotify)
+    {
+        API::CNWSPlayer* observerPlayerObject = static_cast<API::CNWSPlayer*>(server->GetClientObjectByPlayerId(pid, 0));
+
+        if (observerPlayerObject)
+        {
+            // Write a message notifying an object update.
+            message->CreateWriteMessage(0x400, pid, 1);
+
+            // We don't need one for our update.
+            // However, the appearance update is contingent on receiving a pointer which isn't nullptr.
+            API::CLastUpdateObject* lastUpdateObj = reinterpret_cast<API::CLastUpdateObject*>(0xDEADBEEF);
+
+            message->WriteGameObjUpdate_UpdateObject(observerPlayerObject, targetObject, lastUpdateObj, 0, 0x400);
+
+            uint8_t* data = nullptr;
+            uint32_t size = 0;
+
+            if (message->GetWriteMessage(&data, &size) && size)
+            {
+                message->SendServerToPlayerMessage(pid, 5, 1, data, size);
+            }
+
+            // Update the player list. This handles updating the hover-over GUI elements too.
+            message->SendServerToPlayerPlayerList_All(observerPlayerObject);
+        }
+
+        
+    }
+
+
+}
+
+
+ArgumentStack Rename::SetPlayerOverrideName(ArgumentStack&& args)
+{
+	 ArgumentStack stack;
+	auto playerObjectID = Services::Events::ExtractArgument<Types::ObjectID>(args);
+    if (auto *pPlayer = player(playerObjectID))
+    {
+        CNWSCreature *pCreature = Globals::AppManager()->m_pServerExoApp->GetCreatureByGameObjectID(playerObjectID);
+        if (!pCreature)
+        {
+            LOG_ERROR("No creature object found for Player ID %x, oidNWSObject %x",
+                pPlayer->m_oidPCObject, playerObjectID);
+			return stack;
+        }
+		
+		const auto sPrefix = Services::Events::ExtractArgument<std::string>(args);
+        const auto newName = Services::Events::ExtractArgument<std::string>(args);
+		const auto sSuffix = Services::Events::ExtractArgument<std::string>(args);
+		
+		const std::string fullDisplayName = sPrefix + newName + sSuffix + "</c>"; //put together the floaty name, escape all colors
```

Comment:
Does `</c>` work without a matching `<c>`?

Reply:
In my testing it does, and not escaping colors makes whatever color is applied to the name apply to the semi colons, but on second thought it might just be better to let the user handle it, since they can add it to the suffix should they want to.